### PR TITLE
Add extra.utils.imshow2 for easier image plotting

### PR DESCRIPTION
**Pull request by JamesWrigley**
I wrote this function ages ago and I'm constantly copying it over to new notebooks, so I figured it might be useful to put here.

### CODE REVIEW

File: src/extra/utils/misc.py [hunk 1]
```diff
@@ -30,3 +30,54 @@ def find_nearest_value(array, value: Any) -> Any:
...
+            default to [plt][matplotlib.pyplot] if none is explicitly passed.
+
+    """
+    if ax is None:
+        # Try to avoid repeatedly importing matplotlib by looking in sys.modules
+        # first.
+        if "matplotlib.pyplot" in sys.modules:
+            plt = sys.modules["matplotlib.pyplot"]
+        else:
+            import matplotlib.pyplot as plt
```
Comment (takluyver) [inline src/extra/utils/misc.py:63]:
```suggestion
        import matplotlib.pyplot as plt
```

Importing something a second time is cheap - it's basically just doing the same lookup in `sys.modules` that you do here. So there's no need to complicate it. :slightly_smiling_face: 

```
In [1]: %time import matplotlib
CPU times: user 257 ms, sys: 235 ms, total: 492 ms
Wall time: 445 ms

In [2]: %time import matplotlib
CPU times: user 21 µs, sys: 0 ns, total: 21 µs
Wall time: 32.2 µs
```

File: src/extra/utils/misc.py [hunk 2]
```diff
@@ -30,3 +30,54 @@ def find_nearest_value(array, value: Any) -> Any:
...
+
+        ax = plt
+
+    imshow_kwargs = dict(interpolation="none")
```
Comment (takluyver) [inline src/extra/utils/misc.py:67]: Should the docstring mention this too?

File: src/extra/utils/misc.py (hunk 1 above)
Comment (JamesWrigley) [reply src/extra/utils/misc.py]: Ah cool, I thought it would re-import the whole thing. Fixed in 7906c54.

**Review by philsmt — APPROVED** [review]
Oh yes, this looks like it may save some time regularly. Not a fan of the name, but I also don't got any other ideas.

Nothing to complain, just an idea: I regularly use `norm=LogNorm()`  and always forget to add `from matplotlib.colors import LogNorm`. Maybe extend the interface by some `lognorm=True` option?

Still LGTM.

**Comment by JamesWrigley** [comment]
> Nothing to complain, just an idea: I regularly use `norm=LogNorm()` and always forget to add `from matplotlib.colors import LogNorm`. Maybe extend the interface by some `lognorm=True` option?

Good idea, added in b66a3e8 :+1:

**Pull request merged**